pyproject.toml

```diff
@@ -22,3 +22,27 @@ skip = """
 ./tests,\
 ACKNOWLEDGEMENTS,\
 """
+
+[tool.ruff]
+exclude = ["tests"]
+extend-select = ["C9", "PLC"]
+ignore = [
+  "E401",
+  "E402",
+  "E722",
+  "E731",
+  "E741",
+  "F401",
+  "F403",
+  "F405",
+  "F821",
+  "PLC1901",
+]
+line-length = 337
```

Comment:
Shouldn't this match black's line length (99) ?

Reply:
The 99 is aspirational whereas the 337 is actual.

psf/black is unwilling to wrap all lines (comments? docstrings?) so those lines need to be wrapped manually. 

Ideally, both should be 99 once some manual work is done.

Reply:
Hmm then why ruff says "Ruff is compatible with Black out-of-the-box, as long as the line-length setting is consistent between the two." Can you point to an example where ruff complains? There's no way to make it ignore comments for instance (should that be the culprit)?

Reply:
@charliermarsh, @dhruvmanila This question has come up a few times in my travels...

Do you have a better explanation?  Should I create a PR to the ruff docs to improve the explanation there?

Reply:
Hmm, yeah, I'd explain it as: Black makes a best-effort attempt to wrap lines under the specified line length, but doesn't _guarantee_ that all lines will be under the specified line length. But Ruff's rule doesn't discriminate, it just flags all lines over the limit.

You're right that in this particular case, we do arguably "disagree" with Black, depending on how you look at it. Projects with Black enabled could choose to disable this rule entirely if they're happy with Black's line-length handling, though in my _personal_ opinion it still makes sense to enable it. A few examples of code that Black won't wrap, but Ruff _will_ flag:

```py
# Here's a really long comment. Black doesn't wrap comments at all. It wraps docstrings, but it
# doesn't wrap comments.


x = here_is_a_really_really_really_really_really_really_really_really_really_really_long_function_name(
    1, 2, 3
)
```

Reply:
Thanks @charliermarsh  for the explanation.

Sounds like we need to keep this at 337, for now.  Maybe in the future we can tidy up those scripts that go over the 99 length Black limit, so this number here can be reduced.

Reply:
Ok clear.